src/shogun/kernel/AUCKernel.h

```diff
@@ -54,66 +52,19 @@ namespace shogun
 		 */
 		bool setup_auc_maximization();
 
-		/** initialize kernel
-		 *
-		 * @param l features of left-hand side
-		 * @param r features of right-hand side
-		 * @return if initializing was successful
-		 */
-		virtual bool init(std::shared_ptr<Features> l, std::shared_ptr<Features> r);
+		inline virtual EKernelType get_kernel_type() { return K_AUC; }
 
-		/** return what type of kernel we are
-		 *
-		 * @return kernel type AUC
-		 */
-		virtual EKernelType get_kernel_type()
-		{
-			return K_AUC;
-		}
+		inline virtual const char* get_name() const { return "AUCKernel"; }
 
-		/** return the kernel's name
-		 *
-		 * @return name AUC
-		 */
-		virtual const char* get_name() const
-		{
-			return "AUCKernel";
-		}
-
-		/** return feature class the kernel can deal with
-		 *
-		 * @return feature class SIMPLE
-		 */
-		virtual EFeatureClass get_feature_class()
-		{
-			return C_DENSE;
```

Comment:
removing the whole method? Does the superclass already have it?

Reply:
No. Scroll down a bit, I just used inline and made it into one line not removed

Reply:
ah I see, sorry :)